### PR TITLE
allow incluster to accept pass-in config

### PR DESCRIPTION
**Pull request by zshihang**
fixes the e2e tests in kubernetes-client/python/pull/1161.

in_cluster_config with refreshing is mutating the global Configuration class. since all the tests are running in a single process, e2e tests running after the unit tests will use a modified Configuration.

### CODE REVIEW

**Comment by zshihang** [comment]
travis has passed: https://travis-ci.org/github/kubernetes-client/python-base/builds/686358803

File: config/incluster_config.py [hunk 1]
```diff
@@ -35,30 +35,33 @@ def _join_host_port(host, port):
 
 
 class InClusterConfigLoader(object):
-
-    def __init__(self, token_filename,
-                 cert_filename, environ=os.environ):
+    def __init__(self,
+                 token_filename,
+                 cert_filename,
+                 try_refresh_token,
```
Comment (roycaihw) [inline config/incluster_config.py:41]: This is a breaking change. Please add a default value

File: config/incluster_config.py [hunk 2]
```diff
@@ -35,30 +35,33 @@ def _join_host_port(host, port):
...
+                 environ=os.environ):
         self._token_filename = token_filename
         self._cert_filename = cert_filename
         self._environ = environ
+        self._try_refresh_token = try_refresh_token
         self._token_refresh_period = datetime.timedelta(minutes=1)
 
-    def load_and_set(self, refresh_token=True):
+    def load_and_set(self, client_configuration):
```
Comment (roycaihw) [inline config/incluster_config.py:49]: Can we add a separate method and keep the existing one backwards-compatible?

File: config/incluster_config.py (hunk 1 above)
Comment (zshihang) [reply config/incluster_config.py]: done

File: config/incluster_config.py (hunk 2 above)
Comment (zshihang) [reply config/incluster_config.py]: added default value

File: config/incluster_config.py [hunk 3]
```diff
@@ -35,30 +35,39 @@ def _join_host_port(host, port):
...
         self._cert_filename = cert_filename
         self._environ = environ
+        self._try_refresh_token = try_refresh_token
         self._token_refresh_period = datetime.timedelta(minutes=1)
 
-    def load_and_set(self, refresh_token=True):
+    def load_and_set(self, client_configuration=None):
+        try_set_default = False
+        if client_configuration is None:
+            config = type.__call__(Configuration)
```
Comment (roycaihw) [inline config/incluster_config.py:52]: s/config/client_configuration?

Comment (zshihang) [reply config/incluster_config.py]: good catch

**Pull request merged**